src/main/java/seedu/address/model/property/UniqueList.java

```diff
@@ -0,0 +1,138 @@
+package seedu.address.model.property;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.property.exceptions.DuplicateListableException;
+import seedu.address.model.property.exceptions.ListableNotFoundException;
+
+/**
+ * A abstract list that enforces uniqueness between its elements and does not allow nulls.
+ * A {@code Listable} item is considered unique by comparing using {@code Listable#isSameListable(Listable)}.
+ * As such, adding and updating of listables uses Listable#isSameListable(Listable) for equality
+ * so as to ensure that the property being added or updated is unique in terms of identity in the
+ * UniqueListableList. However, the removal of a property uses Listable#equals(Object) so
```

Comment:
Similarly, replace property with element. Ditto for line 20 as well (and other cases).